crypto/crypto/src/unit_tests/noise_test.rs

```diff
@@ -9,8 +9,36 @@ use crate::{
     x25519, Uniform as _,
 };
 
+use crate::ed25519::{Ed25519PrivateKey, Ed25519PublicKey};
+use ed25519_dalek::SecretKey;
 use rand::SeedableRng;
 use serde::*;
+use std::convert::TryFrom;
+
+#[test]
+fn convert_from_ed25519_publickey() {
+    let mut rng = ::rand::rngs::StdRng::from_seed(TEST_SEED);
+    let ed25519_private_key = Ed25519PrivateKey::generate(&mut rng);
+    let ed25519_public_key: Ed25519PublicKey = Ed25519PublicKey::from(&ed25519_private_key);
+    let x25519_public_key = x25519::PublicKey::try_from(&ed25519_public_key);
+    assert!(x25519_public_key.is_ok());
+
+    // Let's try to reconstruct an x25519 private key from the ed25519 private key.
+    let ed25519_secretkey = SecretKey::from_bytes(&ed25519_private_key.to_bytes()).unwrap();
+    let expanded_ed25519_privatekey: ed25519_dalek::ExpandedSecretKey =
+        ed25519_dalek::ExpandedSecretKey::from(&ed25519_secretkey);
+
+    // Note that the first 32 bytes of the ExpandedSecretKey correspond to the actual private key.
+    let mut expanded_keypart = [0u8; 32];
+    let bytes = &expanded_ed25519_privatekey.to_bytes()[..32];
+    expanded_keypart.copy_from_slice(bytes);
+    let x25519_privatekey = x25519::PrivateKey::from(expanded_keypart);
```

Comment:
could we hide this in a function within Ed25519PrivateKey? I'd like to not expose this outside of this crate.

Reply:
Do you prefer a  try_from inside x25519 (as I did for public key) or because it's a private key (sensitive) you prefer this being a function of Ed25519PrivateKey?

Reply:
I mean... this API comes with its risks, I would say try_from is good enough with the caveat stating that this is intended to only be used for "cases where KMS (key management service) can only store Ed25519 keys (i.e., a Vault or HSM), but we use it for x25519 noise (DH) protocol communication." or the like.